opentelemetry-user-events-trace/examples/basic-trace.rs

```diff
@@ -0,0 +1,51 @@
+//! run with `$ cargo run --example basic-trace --all-features`
+
+use opentelemetry::global;
+use opentelemetry::trace::Span;
+use opentelemetry::trace::Tracer;
+use opentelemetry_sdk::trace::SdkTracerProvider;
+use opentelemetry_user_events_trace::UserEventsTracerProviderBuilderExt;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::{thread, time::Duration};
+
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt, EnvFilter, Layer};
+
+fn init_tracer() -> SdkTracerProvider {
+    let provider = SdkTracerProvider::builder()
+        .with_user_event_exporter("myprovider")
+        .build();
+    global::set_tracer_provider(provider.clone());
+    provider
+}
+
+fn main() {
+    let filter_fmt = EnvFilter::new("info").add_directive("opentelemetry=debug".parse().unwrap());
+    let fmt_layer = tracing_subscriber::fmt::layer().with_filter(filter_fmt);
+    tracing_subscriber::registry().with(fmt_layer).init(); // Temporary subscriber active for this function
```

Comment:
what is temporary subscriber?

Reply:
Removed the comment, was copy-paste from logs.